yaramo/signal.py

```diff
@@ -52,8 +71,38 @@ class SignalState(Enum):
     hp2 = 2
     ks1 = 3
     ks2 = 4
-    sh1 = 5
-    sh2 = 6
+    sh0 = 5
+    sh1 = 6
+    sh2 = 7
+    ne2 = 8
+    zs1 = 9
+    zs2 = 10
+    zs2v = 11
+    zs3 = 12
+    zs3v = 13
+    zlo = 14
+    lf7 = 15
+    ra10 = 16
+    ra12 = 17
+    ms_ws_rt_ws = 18
+    ms_ge_d = 19
+
+    @classmethod
+    def get_state_by_string(cls, state_string: str):
+        states = dict([(e.name, e) for e in SignalState])
```

Comment:
Brackets not needed, dict comprehension possible:
`states = {e.name: e for e in SignalState}`
There is also `SignalState.__members__` which could do this for us.

Reply:
Nice 👍